include/envoy/thread/thread.h

```diff
@@ -61,7 +61,7 @@ class SCOPED_LOCKABLE TryLockGuard {
    *
    * @param lock the mutex.
    */
-  TryLockGuard(BasicLockable& lock) EXCLUSIVE_TRYLOCK_FUNCTION(true)
+  TryLockGuard(BasicLockable& lock) EXCLUSIVE_LOCK_FUNCTION(lock)
```

Comment:
That works for hot_restart_impl.cc because it doesn't have any GUARDED_BY variables -- I'm not sure which variables should be guarded by the init_lock_ -- no variables are consistently protected by it currently.

However if I make this testcase:
```
#include <string>

#include "common/common/thread.h"

#include "gtest/gtest.h"

namespace Envoy {
namespace Thread {

class ThreadTest : public testing::Test {
 protected:
  ThreadTest() : a_(0), b_(0) {}
  int a_ GUARDED_BY(a_mutex_);
  MutexBasicLockable a_mutex_;
  int b_;
};

TEST_F(ThreadTest, TestLockGuard) {
  LockGuard lock(a_mutex_);
  EXPECT_EQ(1, ++a_);
}

TEST_F(ThreadTest, TestOptionalLockGuard) {
  OptionalLockGuard lock(nullptr);
  EXPECT_EQ(1, ++b_);
}

TEST_F(ThreadTest, TestReleasableLockGuard) {
  ReleasableLockGuard lock(a_mutex_);
  EXPECT_EQ(1, ++a_);
  lock.release();
}

TEST_F(ThreadTest, TestTryLockGuard) {
  TryLockGuard lock(a_mutex_);
  if (lock.tryLock()) {
    EXPECT_EQ(1, ++a_);
  }
}

} // namespace Thread
} // namespace Envoy
```
with TryLockGuard unannotated, I get this compile error:
```
[0 / 5] [-----] BazelWorkspaceStatusAction stable-status.txt
ERROR: /usr/local/google/home/jmarantz/git3/envoy/test/common/common/BUILD:139:1: C++ compilation of rule '//test/common/common:thread_test_lib' failed (Exit 1)
test/common/common/thread_test.cc:37:20: error: writing variable 'a_' requires holding mutex 'a_mutex_' exclusively [-Werror,-Wthread-safety-analysis]
    EXPECT_EQ(1, ++a_);
                   ^
1 error generated.
```